src/controller/OnOffCluster.cpp

```diff
@@ -22,39 +22,41 @@
 namespace chip {
 namespace Controller {
 
-// TODO: Find a way to calculate maximum message length for clusters
-//       https://github.com/project-chip/connectedhomeip/issues/965
-constexpr uint16_t kMaxOnOffMessageLength = 64;
-
 CHIP_ERROR OnOffCluster::On(Callback::Callback<> * onCompletion)
 {
-    return SendCommand(encodeOnOffClusterOnCommand, kMaxOnOffMessageLength, onCompletion);
+    System::PacketBufferHandle payload = encodeOnOffClusterOnCommand(mEndpoint);
+    return SendCommand(std::move(payload), onCompletion);
```

Comment:
If desired, you could even:
```
  return sendCommand(encodeOnOffClusterOnCommand(mEndpoint), onCompletion);
```
but either way is fine.

Reply:
That's initially what I was targeting but for command with a bunch of parameters it be hard to read. So I guess I would stay with the 2 lines version, except if someone has strong opinion about that.